core/controllers/cron.py

```diff
@@ -86,6 +89,16 @@ def get(self):
         email_services.send_mail_to_admin(email_subject, email_message)
 
 
+class AppendLastDashboardStatsToList(base.BaseHandler):
+    """Handler for appending dashboard stats to a list."""
+
+    @require_cron_or_superadmin
+    def get(self):
+        """Handles GET requests."""
+        for user_model in user_models.UserSettingsModel.get_all():
```

Comment:
It does not seem realistic to load all the thousands of instances of UserSettingsModel and iterate through them in a cron request. You may need to kick off a one-off mapreduce, or something.